Add support for PR and Issue templates to QMK_firmware GitHub

drashna opened the pull request:
* Adds templates for issues and PRs
* Add support document for GitHub
* Add support link to docs

mechmerlin commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,15 @@
+- [ ] New behavior
+
+## Description 
+A few sentances describing what it is that you'd like to see. Additional information (such as links to spec sheets, licensing info, other related issues or PR's, etc) would be helpful. 
```
While the template is nice for us, I worry it may intimidate people who may have never filed a bug or wandered into Github before. Perhaps we should put in some verbage saying that its optional?

drashna replied:
Absolutely. Maybe we should add a blank issue, as well?

drashna commented:
Also, would something like the following bet better for the pull request template? 
```md
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
```

mechmerlin commented on docs/support.md:
```diff
@@ -0,0 +1,43 @@
+
+## What if I found a bug? 
+
+Then please open an issue, and if you know how to fix it, open up a Pull Request on GitHub with the fix.
```
```suggestion
Then please open an [issue](https://github.com/qmk/qmk_firmware/issues/new), and if you know how to fix it, open up a Pull Request on GitHub with the fix.
```
